zookeeper-server/src/main/java/org/apache/zookeeper/server/watch/WatchManager.java

```diff
@@ -74,22 +74,10 @@ public synchronized boolean addWatch(String path, Watcher watcher, WatcherMode w
             return false;
         }
 
-        Set<Watcher> list = watchTable.get(path);
-        if (list == null) {
-            // don't waste memory if there are few watches on a node
-            // rehash when the 4th entry is added, doubling size thereafter
-            // seems like a good compromise
-            list = new HashSet<>(4);
-            watchTable.put(path, list);
```

Comment:
We should still retain these annotations? others lgtm

Reply:
Maybe It's not necessary